src/OrleansServiceBus/Providers/Streams/EventHub/IEventHubQueueCacheFactory.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Orleans.Runtime;
+using Orleans.Streams;
+
+namespace Orleans.ServiceBus.Providers
+{
+    /// <summary>
+    /// Factory responsible for creating a message cache for an EventHub partition.
+    /// </summary>
+    public interface IEventHubQueueCacheFactory
```

Comment:
Fair point.  The cost of this pattern is that it requires one to subclass the adapter factory, but I guess that's kinda its purpose.  My hope is to replace all of this with a builder some day.
One can dream!? :)